src/main/java/rx/internal/operators/OperatorConcat.java

```diff
@@ -179,6 +187,14 @@ void subscribeNext() {
                 }
             }
         }
+        
+        void produced(long c) {
+            arbiter.produced(c);
+            AtomicLong requestedField = requested;
+            if (requestedField.get() != Long.MAX_VALUE) {
+                requestedField.addAndGet(-c);
```

Comment:
I've a very narrow focus on this method so I could well be missing something but normally once `requested` is set to Long.MAX_VALUE we don't reduce its value and @zsxwing has indicated this possibility. I don't think it should be best endeavours as in I would expect to see a CAS loop here. Is this a bit fussy? Do we do this in lots of places already? (I thought a lot had been cleaned up).